src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -300,6 +300,30 @@ public Dataset getDatasetData(Program program, UUID experimentId, UUID datasetId
         log.debug("fetching dataset: " + datasetId + " for experiment: " + experimentId + ".  including stats: " + stats);
         log.debug("fetching observationUnits for dataset: " + datasetId);
         List<BrAPIObservationUnit> datasetOUs = ouDAO.getObservationUnitsForDataset(datasetId.toString(), program);
+
+        //Add years to the addition_info elements
+        Map<String, Integer> studyDbId2Year = new HashMap<>();  // used to prevent the same study from being fetched repeatedly.
```

Comment:
I think the naming of `studyDbId2Year` could be more clear, for example `yearByStudyDbId` would indicate that you can use the map to look up a year by a studyDbId and would also have parallelism with other maps in bi-api that are named `{value}By{key}`.

Reply:
Done (and thank you)